### DISCUSSION TITLE
When waiting for too long in a queue line, peep should not want to go back on the same ride

### DISCUSSION THREAD

**Discussion by frutiemax** [OpenRCT2/OpenRCT2#15735] [Feature requests]
I've encountered this bug many times : basically, when a peep waits for too long i.e. the happiness goes to its minimum, the peep should either leave the park or go do something else. However, I've seen the peep go back into the same queue line he raged for ages.

**Comment by ocalhoun6** [comment]
A simple way to implement this might be to take the peep code that's executed when a peep encounters a queue line for a ride, and have that code check the peep's recent thoughts. If "I've been waiting in line for [same ride as the peep is now encountering] ages" is in the list, then have the peep refuse to ride that ride.

That will make the peep's refusal to reenter a too-long line pretty temporary ... but it _should_ keep him out of the same ride's queue long enough for him to wander off somewhere else in most cases. By the time that thought leaves his recent thoughts, he'll _probably_ be far enough away that he's unlikely to encounter the same queue line again.

But in some ways, temporary is good. You don't want a peep to give up on that ride _forever_. If he comes back much later, the queue might be shorter.

(And maybe do the same for, "I want to get off [current ride]"? If a peep has been stuck on a ride long enough to be thinking that, they probably wouldn't want to get right back on the same ride.)

---------------------------------------------------------------------------------

Maybe there could even be a little self-regluation of nausea? Perhaps a peep could be less likely to get in line for a high-nausea ride if he's already feeling sick? That's more of a drastic change, though, because it could cause a pretty significant difference in peep behavior, resulting in less vomit and also less popularity of high-nausea rides.

Some self-regulation on wait times might make sense also. The game already keeps track of average wait time for each ride. Peeps could look at that and be less likely to choose the ride if the wait time is high. (Which would pretty naturally reflect the behavior of real-world people.)

**Comment by frutiemax** [comment]
I was also thinking for a fix that is very temporary i.e. checks the last action, if last action = waiting in line for that ride, then don't go back for that ride. I wouldn't fix the guest being sick going to a nausea because this can happen in the real world (and it's funny).

**Comment by Ruedii** [comment]
Peeps aren't that smart, just like real park guests.

They will go back to the same ride and gripe about the line being too long again.

**Comment by ocalhoun6** [comment]
While I realize the hubris of postulating a limit to stupidity, I really doubt that it's common for real theme park guests to get so fed up waiting in line that they leave the line, then _immediately_ get back in the end of that same line.

Even as little as a few minutes later, sure. I could see that. Then they might be thinking 'oh, maybe the line will go faster this time'. But to just walk back to the end of the line and get in line there? Almost nobody is stupid enough to give up their place in line for no reason.
